File: frontend/src/types/doctorlessArea.ts
export interface DoctorlessArea {
  doctorless_city_id: number;
  municipality_id: number;
  municipality_name: string;
  city_raw_id: number;
  city_name: string;
  doctorless_flag: boolean;
  location: {
    latitude: number;
    longitude: number;
  };
}

export interface DoctorlessApiResponse {
  meta: {
    limit: number;
    count: number;
  };
  data: DoctorlessArea[];
}

File: backend/app/schemas/doctorless_area.py
from pydantic import BaseModel


class DoctorlessAreaLocationResponse(BaseModel):
    latitude: float | None
    longitude: float | None


class DoctorlessAreaResponse(BaseModel):
    doctorless_city_id: int
    municipality_id: int
    municipality_name: str | None
    city_raw_id: int
    city_name: str | None
    doctorless_flag: bool | None
    location: DoctorlessAreaLocationResponse


class DoctorlessAreaListMetaResponse(BaseModel):
    limit: int
    count: int


class DoctorlessAreasResponse(BaseModel):
    meta: DoctorlessAreaListMetaResponse
    data: list[DoctorlessAreaResponse]
